x-pack/plugins/reporting/server/routes/lib/jobs_query.ts

```diff
@@ -33,7 +32,25 @@ function getSearchBody(body: SearchRequest['body']): SearchRequest['body'] {
   };
 }
 
-export function jobsQueryFactory(reportingCore: ReportingCore) {
+export type ReportContent = Pick<ReportSource, 'status' | 'jobtype' | 'output'> & {
+  payload?: Pick<ReportSource['payload'], 'title'>;
+};
+
+interface JobsQueryFactory {
+  list: (
+    jobTypes: string[],
+    user: ReportingUser,
+    page: number,
+    size: number,
+    jobIds: string[] | null
+  ) => Promise<ReportApiJSON[]>;
+  count: (jobTypes: string[], user: ReportingUser) => Promise<number>;
+  get: (user: ReportingUser, id: string) => Promise<ReportApiJSON | void>;
+  getContent: (user: ReportingUser, id: string) => Promise<ReportContent | void>;
+  delete: (deleteIndex: string, id: string) => Promise<ApiResponse<DeleteResponse>>;
```

Comment:
Can we declare those as methods rather than function properties? That would be more correct, and we will be able to override them when implementing the interface?

See [TypeScript documentation](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-6.html).
> The stricter checking applies to all function types, except those originating in method or constructor declarations. Methods are excluded specifically to ensure generic classes and interfaces (such as Array<T>) continue to mostly relate covariantly.

Reply:
93dadad8fb1

Reply:
Thanks for this catch - it was an accident :)